src/operator/controllers/istiopolicy/policy_manager.go

```diff
@@ -316,14 +318,32 @@ func shouldUpdateStatus(intents v1alpha2.ClientIntents, currentName string, curr
 	return false
 }
 
-func (c *AdminImpl) createOrUpdatePolicies(
+func (c *PolicyManagerImpl) createOrUpdatePolicies(
 	ctx context.Context,
 	clientIntents *v1alpha2.ClientIntents,
 	clientServiceAccount string,
 	existingPolicies v1beta1.AuthorizationPolicyList,
-) (goset.Set[PolicyID], error) {
+) (*goset.Set[PolicyID], error) {
 	updatedPolicies := goset.NewSet[PolicyID]()
+	createdAnyPolicies := false
 	for _, intent := range clientIntents.GetCallsList() {
+		shouldCreatePolicy, err := protected_services.ShouldCreateNetworkPoliciesDueToProtectionOrDefaultState(
+			ctx, c.client, intent.GetServerName(), intent.GetServerNamespace(clientIntents.Namespace), c.enforcementDefaultState)
+		if err != nil {
+			return nil, err
+		}
+
+		if !shouldCreatePolicy {
+			logrus.Infof("Enforcement is disabled globally and server is not explicitly protected, skipping network policy creation for server %s in namespace %s", intent.GetServerName(), intent.GetServerNamespace(clientIntents.Namespace))
+			c.recorder.RecordNormalEventf(clientIntents, consts.ReasonEnforcementDefaultOff, "Enforcement is disabled globally and called service '%s' is not explicitly protected using a ProtectedService resource, network policy creation skipped", intent.Name)
+			continue
+		}
+
+		if !c.enableIstioPolicyCreation {
```

Comment:
Why is this a separate flag ?

Reply:
This is the flag that disabled Istio policy creation, like the one we have for netpols and Kafka